applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_tests/adjoint_shell_structure_3d3n/linear_shell_test_nodal_disp_adjoint_parameters.json

```diff
@@ -23,13 +23,16 @@
             "response_type"     : "adjoint_nodal_displacement",
             "use_kratos"        : true,
             "gradient_mode"     : "semi_analytic",
+            "step_size"         : 1.0e-6,
+            "traced_node_id"       : 5,
```

Comment:
I don't really see another option to refer to a certain node/element than using the ID in the input file:
 * MP reordering: if this happens, it is important that the node object/pointer gets extracted beforehands. 
 * slow: this is done for one node only, absolutely not critical
 * MPI also there i expect it is a matter of when the node is referenced. In the end also elements are created using node ids, if that is working, i see no reason why it should not be possible to reference it in the response too.

One potential alternative could be to create a submodelpart with just one node inside, so at least we would give the responsibility for all the above things to the model part IO... 

But i don't like that idea a lot, because it suggests that you could use more that one node, which you don't.

Reply:
> MP reordering: if this happens, it is important that the node object/pointer gets extracted beforehands.

I don't think this is possible, the reordering happens at the same time

> 
>  slow: this is done for one node only, absolutely not critical

ok makes sense

> 
> MPI also there i expect it is a matter of when the node is referenced. In the end also elements are created using node ids, if that is working, i see no reason why it should not be possible to reference it in the response too.

same here, the reordering happens at reading time
> 
> 
> One potential alternative could be to create a submodelpart with just one node inside, so at least we would give the responsibility for all the above things to the model part IO...

This seems like the best solution to me, since the responsibility is taken from the user (and put where it actually belongs). 
> 
> But i don't like that idea a lot, because it suggests that you could use more that one node, which you don't.

You could easily check if the submodelpart has only one node.

After all I am just pointing out potential bottlenecks, decision is up to you

Reply:
Why not give the spatial coordinates instead of node id? We already do this for point output.

Reply:
I don't like the idea with the spatial coordinates for several reasons
* its more cumbersome for the user
* more error prone, what about tolerances (also not hard to solve, but still potential source of errors)
* more expensive - node search is needed (even if here it is not critical)

I think going for the model part solution can be a long term solution. But definitely in a separate PR.

> After all I am just pointing out potential bottlenecks, decision is up to you

@philbucher I know and I really appreciate that!

Reply:
I opened a new issue: #3492

Reply:
I don't agree that it is more cumbersome for the user or error prone as long as it works for any point in the domain, not just the nodes. As pointed out this is already done for point output for a while and we don't have any issues with it. Actually, trying to find a node id would be more annoying imo.